src/js/jsx/sections/pages/LayerFace.jsx

```diff
@@ -40,18 +40,43 @@ define(function (require, exports, module) {
         system = require("js/util/system"),
         strings = require("i18n!nls/strings");
 
-    var LayerFace = React.createClass({
-        mixins: [FluxMixin],
+    /*
+     * Function for checking whether React component should update
+     * Passed to Droppable composed component in order to save on extraneous renders
+     *
+     * @param {Object} nextProps - Next set of properties for this component
+     * @return {boolean}
+     */
+    var shouldComponentUpdate = function (nextProps) {
+        // Drag states
+        if (this.props.dragTarget !== nextProps.dragTarget ||
+            this.props.dropAbove !== nextProps.dropAbove ||
+            this.props.dragPosition !== nextProps.dragPosition ||
+            this.props.dragStyle !== nextProps.dragStyle ||
+            this.props.dropTarget !== nextProps.dropTarget) {
+            return true;
+        }
+        
+        // Face change
+        if (!Immutable.is(this.props.layer.face, nextProps.layer.face)) {
+            return true;
+        }
 
-        shouldComponentUpdate: function (nextProps) {
-            return !Immutable.is(this.props.layer.face, nextProps.layer.face) ||
-                this.props.dragTarget !== nextProps.dragTarget ||
-                this.props.dropAbove !== nextProps.dropAbove ||
-                this.props.dragPosition !== nextProps.dragPosition ||
-                this.props.dragStyle !== nextProps.dragStyle ||
-                this.props.dropTarget !== nextProps.dropTarget;
-        },
+        // Deeper selection changes
+        var document = this.props.document,
+            childOfSelection = document.layers.hasSelectedAncestor(this.props.layer),
+            nextChildOfSelection = nextProps.document.layers.hasSelectedAncestor(nextProps.layer);
+            
+        if (childOfSelection || nextChildOfSelection) {
```

Comment:
It's awkward to do this with JavaScript, but there _is_ an opportunity here to potentially avoid one of the above method calls be rewriting this guard.